lib/clustering/mean_shift.h

```diff
@@ -9,19 +9,39 @@
 #include "clustering.h"
 #include <vector>
 #include "generic/cluster.h"
+#include <Eigen/Core>
 
 namespace MouseTrack {
 
```

Comment:
A short description of the algorithm would be nice, especially since you keep referencing "mean-shift iteration" at other places, but never define the term.